platforms/core-runtime/launcher/src/integTest/groovy/org/gradle/launcher/daemon/server/scaninfo/DaemonScanInfoIntegrationSpec.groovy

```diff
@@ -209,15 +218,13 @@ class DaemonScanInfoIntegrationSpec extends DaemonIntegrationSpec {
         """
         task waitForExpiration {
             doFirst {
-                if (!latch.await(2, TimeUnit.SECONDS)) {
-                    throw new GradleException("Timed out waiting for expiration event")
-                }
+                Thread.sleep(2000)
```

Comment:
❓ This isn't exactly equivalent to what we had before, because we no longer block until the expiration listener fires. I suppose it can bring flakiness when the task (and thus the build) completes earlier than the listener. In this configuration I don't see how the server is different to just printing something from the listener.

Maybe a better solution would be to do something like:
```
server.expectConcurrent("notifyOnUnhealthy", "runTask")
...

doFirst {
  ${server.callFromTaskAction("runTask")}
}
```

The listener stays the same way it is in this PR.

This way we ensure that the build runs long enough to deliver the expiration event.